packages/common/client-utils/src/layerCompat.ts

```diff
@@ -0,0 +1,128 @@
+/*!
+ * Copyright (c) Microsoft Corporation and contributors. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+/**
+ * Result of a layer compatibility check - whether a layer is compatible with another layer.
+ * @internal
+ */
+export type LayerCompatCheckResult =
+	| { readonly isCompatible: true }
+	| {
+			readonly isCompatible: false;
+			/**
+			 * Whether the generation of the layer is compatible with the other layer.
+			 */
+			readonly isGenerationCompatible: boolean;
+			/**
+			 * The features that are required by the layer but are not supported by the other layer. This will only
+			 * be set if there are unsupported features.
+			 */
+			readonly unsupportedFeatures: readonly string[] | undefined;
+	  };
+
+/**
+ * @internal
+ */
+export const ILayerCompatDetails: keyof IProvideLayerCompatDetails = "ILayerCompatDetails";
```

Comment:
I thought we were done with this pattern...?  Who is ever querying for this?  I guess I'll find out later.